docs/developer-guide/etendo-classic/how-to-guides/how-to-create-build-validations-and-module-scripts.md

```diff
@@ -0,0 +1,349 @@
+  * The module script should never fail. If it fails, the build will be stopped, but as it has already been started, **the system will be in an inconsistent state** (for example, all the foreign keys and triggers will be down). The developer must avoid this at all costs if possible. 
+
+###  Introduction to the implementation of module scripts
+
+As with build validations, a module script is a Java class which extends the abstract class `org.openbravo.modulescript.ModuleScript`, and implements the `execute()` method. This execute method will be called by the `update.database` task. This method will include the logic of the module script, here all the operations will be done.
+
+###  Writing the module script
+
+Here goes a very simple **Module script**, which just sets the value of some column to a default in case the column value is null:
+
+```java title="ModuleScriptExample.java"
+public class ModuleScriptExample extends ModuleScript {
+ 
+  public void execute() {
+    try {
+      ConnectionProvider cp = getConnectionProvider();
+      PreparedStatement ps = cp
+          .getPreparedStatement("UPDATE mytable SET mycolumn=0 WHERE mycolumn IS NULL");
+      ps.executeUpdate();
+    } catch (Exception e) {
+      handleError(e);
+    }
+  }
+}
+```
+
+The main two important points to take into account when building the script are:
+
+  * The script should never fail. Failure will leave the user in a very unfriendly situation (with the build stopped in the middle), and **should be avoided at all costs**. In this case, this script will not fail unless the table does not exist. 
+  * The script needs to be designed so that it can be executed an **indefinite number of times**. This example script, for example, can be executed several times without problems (the first time will set all rows which have a null value, and after that, only the new rows which have been inserted before will be modified, but the previously modified rows will not be modified again unless they have the supposedly wrong null value again). 
+
+  
+!!!info
+      The class source files themselves should be inside the module folder, in the folder: `src-util/modulescript/src`. And they should follow the standard **Java package rules**. The folder will not exist if it is the first script, so it will be needed to create it.
+
+
+###  Compiling the module script
+
+Module scripts should be compiled using, at most, the latest supported **JDK version**.
+
+To compile the module script, use the following command:
+
+```bash
+./gradlew compile.modulescript -Dmodule=<javapackage>
+```
+
+!!!info
+      In case of compiling the scripts of **Etendo Core**, the module property needs to be set to `org.openbravo`.
```

Comment:
It is still 'org.openbravo'?

Reply:
Yes. Java package of Core itself is org.openbravo:
![Selección_944](https://github.com/etendosoftware/docs/assets/136735078/b55000f4-cd05-465c-93ed-4ee7386f9842)